hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/KafkaBackendResolver.java

```diff
@@ -0,0 +1,35 @@
+package org.hypertrace.traceenricher.enrichment.enrichers.resolver.backend;
+
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+import org.hypertrace.core.datamodel.Event;
+import org.hypertrace.core.datamodel.shared.StructuredTraceGraph;
+import org.hypertrace.entity.data.service.v1.Entity;
+import org.hypertrace.semantic.convention.utils.messaging.MessagingSemanticConventionUtils;
+import org.hypertrace.traceenricher.enrichment.enrichers.BackendType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class KafkaBackendResolver extends AbstractBackendResolver {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(KafkaBackendResolver.class);
+
+  @Override
+  public Optional<Entity> resolveEntity(Event event, StructuredTraceGraph structuredTraceGraph) {
+    if (!MessagingSemanticConventionUtils.isKafkaBackend(event)) {
+      return Optional.empty();
+    }
+
+    Optional<String> backendURI = MessagingSemanticConventionUtils.getKafkaBackendURI(event);
+
+    if (backendURI.isEmpty() || StringUtils.isEmpty(backendURI.get())) {
+      LOGGER.warn("Unable to infer a kafka backend from event: {}", event);
+      return Optional.empty();
+    }
+
+    Entity.Builder entityBuilder = getBackendEntityBuilder(
+        BackendType.KAFKA, backendURI.get(), event);
```

Comment:
As per specification - https://github.com/open-telemetry/opentelemetry-specification/blob/main/semantic_conventions/trace/messaging.yaml#L49, the URL is in the form - `['tibjmsnaming://localhost:7222', 'https://queue.amazonaws.com/80398EXAMPLE/MyQueue']`, and this method expects `host:port` string. So, use URI/URL parser to form this string in `getKafkaBackendURI`

Reply:
Made this change for sqs backendURI.